wazuh/wazuh-qa
# Fix `test_synchronization` system test

# Description

Change the execution of queries using the correct script (adding it to the manager during provisioning). Fix the test that gave an error for not using the json format and adjust the condition so that the test passes or fails.

## Testing Performed

### Steps to Test on Local

First you have to download the code and the branch with the fix, then you just have to execute the commands in order to launch the test.

```console
python3 -m pip install -r requirementes.txt
cd deps/wazuh_testing && python3 setup.py install
cd ../../tests/system/provisioning/one_manager_agent/ && ansible-playbook -i inventory.yml destroy.yml
ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"package_repository":"packages-dev", "repository": "pre-release", "package_version": "4.8.0", "package_revision": "1"}'
cd ../../ && python3 -m pytest -vvrxP test_fim/test_synchronization/test_synchronization.py 
```

When provisioning you have to ensure that you take the packages from the correct bucket.

<details>
  <summary>Results</summary>
  <br/>

```console
------------------------------ Captured log call -------------------------------
DEBUG    wazuh_testing:system_monitoring.py:78 Add new file composer process for wazuh-agent1 and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:82 Add new file monitor process for wazuh-agent1 and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:78 Add new file composer process for wazuh-manager and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:82 Add new file monitor process for wazuh-manager and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:185 Cleaning temporal files...
DEBUG    wazuh_testing:system_monitoring.py:174 Checking results...
DEBUG    wazuh_testing:system_monitoring.py:180 Received from wazuh-agent1 the expected message: True
DEBUG    wazuh_testing:system_monitoring.py:180 Received from wazuh-manager the expected message: True
DEBUG    wazuh_testing:system_monitoring.py:78 Add new file composer process for wazuh-manager and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:82 Add new file monitor process for wazuh-manager and path: /var/ossec/logs/ossec.log
DEBUG    wazuh_testing:system_monitoring.py:185 Cleaning temporal files...
DEBUG    wazuh_testing:system_monitoring.py:174 Checking results...
DEBUG    wazuh_testing:system_monitoring.py:180 Received from wazuh-manager the expected message: True
======================== 6 passed in 201.24s (0:03:21) =========================
```

</details>

I have run the test several times to check the functionality

```console
for i in {1..10}; do python3 -m pytest -vvrxP test_fim/test_synchronization/test_synchronization.py; done >> results
```

All 10 test runs have passed correctly. This is the result, in HTML format, of the execution of the tests [report.zip](https://github.com/wazuh/wazuh-qa/files/14561170/report.zip)
